codice/ddf
# [2.19.x] G-9817 G-9818 G-9819 Fix saving search forms that use Text MATCHCASE, Text NEAR, and Numeric RANGE operators

#### What does this PR do?
Prior to this change, trying to save a search form that used a Text MATCHCASE, Text NEAR, or Numeric RANGE operator would fail with a red error banner on the UI. This fixes the issue. 

#### Who is reviewing it? 
#### Ask 2 committers to review/merge the PR and tag them here.
@brianfelix
@cassandrabailey293
@millerw8
@mojogitoverhere

#### How should this be tested?
#### Testing "MATCHCASE"
- Open search forms and start a new form.
- Pick any Text attribute, such as `anyText`, and pick the `MATCHCASE` operator.
- Ensure the form saves.
- Ensure refreshing the page doesn't result in errors and the form gets resurrected correctly. 
- Repeat the process with the exact same predicate but have it embedded in a logical operator of some kind (and/or).
- Open Solr, find the form metacard, grab the XML payload in the `ui.template-filter` attribute and ensure when prettified it looks like this:
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
<Filter xmlns=\"http://www.opengis.net/fes/2.0\" xmlns:ns2=\"http://www.opengis.net/ows/1.1\" xmlns:ns3=\"http://www.w3.org/1999/xlink\">
	<Function name=\"custom.preds.like\">
		<Literal>anyText</Literal>
		<Literal>english</Literal>
	</Function>
</Filter>
```
- Also verify the XML for the form with the logical operator, which should look something like this:
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
<Filter xmlns=\"http://www.opengis.net/fes/2.0\" xmlns:ns2=\"http://www.opengis.net/ows/1.1\" xmlns:ns3=\"http://www.w3.org/1999/xlink\">
	<And>
		<Function name=\"custom.preds.like\">
			<Literal>anyText</Literal>
			<Literal>english</Literal>
		</Function>
		<Intersects>
			<ValueReference>anyGeo</ValueReference>
			<Literal>POLYGON((-112.236664 33.655016,-111.999747 33.607969,-112.228186 33.411716,-112.236664 33.655016))</Literal>
		</Intersects>
	</And>
</Filter>
```

#### Testing "NEAR"
- Open search forms and start a new form.
- Pick any Text attribute, such as `anyText`, and pick the `NEAR` operator.
- Ensure the form saves.
- Ensure refreshing the page doesn't result in errors and the form gets resurrected correctly. 
- Repeat the process with the exact same predicate but have it embedded in a logical operator of some kind (and/or).
- Open Solr, find the form metacard, grab the XML payload in the `ui.template-filter` attribute and ensure when prettified it looks like this:
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
<Filter xmlns=\"http://www.opengis.net/fes/2.0\" xmlns:ns2=\"http://www.opengis.net/ows/1.1\" xmlns:ns3=\"http://www.w3.org/1999/xlink\">
	<PropertyIsEqualTo>
		<Function name=\"proximity\">
			<Literal>anyText</Literal>
			<Literal>2</Literal>
			<Literal>english</Literal>
		</Function>
		<Literal>true</Literal>
	</PropertyIsEqualTo>
</Filter>
```
- Also verify the XML for the form with the logical operator, which should look something like this:
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
<Filter xmlns=\"http://www.opengis.net/fes/2.0\" xmlns:ns2=\"http://www.opengis.net/ows/1.1\" xmlns:ns3=\"http://www.w3.org/1999/xlink\">
	<And>
		<PropertyIsEqualTo>
			<Function name=\"proximity\">
				<Literal>anyText</Literal>
				<Literal>2</Literal>
				<Literal>english</Literal>
			</Function>
			<Literal>true</Literal>
		</PropertyIsEqualTo>
		<PropertyIsLike wildCard=\"%\" singleChar=\"_\" escapeChar=\"\\\">
			<ValueReference>ext.location-be-number</ValueReference>
			<Literal>B89*</Literal>
		</PropertyIsLike>
	</And>
</Filter>
```

#### Testing "RANGE"
- Open search forms and start a new form.
- Pick any Numeric attribute, such as `Band Count`, and pick the `RANGE` operator.
- Ensure the form saves.
- Ensure refreshing the page doesn't result in errors and the form gets resurrected correctly. 
- Repeat the process with the exact same predicate but have it embedded in a logical operator of some kind (and/or).
- Open Solr, find the form metacard, grab the XML payload in the `ui.template-filter` attribute and ensure when prettified it looks like this:
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
```
- Also verify the XML for the form with the logical operator, which should look something like this:
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
```

#### Everything in one form
Finally, here's an example from Solr of a compound search form leveraging all predicates fixed as part of this PR: 
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
```

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
